sdk/core/core/inc/az_http_request_builder.h

```diff
@@ -123,6 +124,23 @@ AZ_NODISCARD az_result az_http_request_builder_append_header(
     az_span const key,
     az_span const value);
 
+/**
+ * @brief Adds path to url request.
+ * For instance, if url in request is `http://example.net` and this function is called with path
+ * equals to `test`, then request url will be updated to `http://example.net/test`.
+ * Use this function to create a path after url.
+ *
+ * Note: This function will work as long as no query parameter has been added to url. Otherwise
```

Comment:
This is fine for now but this restriction should really be removed. I'd like to be able to add a path and have all the QPs shift down in memory.

Reply:
creating issue to handle this eventually https://github.com/Azure/azure-sdk-for-c/issues/237.

I didn't want to expend much time into this. 
I have some ideas on my mind about how to do this, but I would like to hear more ideas.
I have two proposals:
1)  Use temp stack buffer
    -  copy QPs to temp buffer in stack
    -  write path
    -  copy back QPs from temp buffer after path

Reply:
No temporary buffers. First, make sure there is enough space and if not, fail. Then, determine split point and shift ?QPs down by path-#-bytes. Then, copy path bytes into opened-up space.

Reply:
Roger that.
I was afraid of overriding QPs value while shifting right, but just realized I can do it from right to left and that works 👍